jenkins-scripts/docker/lib/docker_generate_dockerfile.bash

```diff
@@ -33,24 +33,32 @@ export APT_PARAMS=
 
 GZDEV_DIR=${WORKSPACE}/gzdev
 GZDEV_BRANCH=${GZDEV_BRANCH:-master}
+if [[ "$ghprbSourceBranch" =~ matching_branch\/ ]]; then
+  GZDEV_TRY_BRANCH=$ghprbSourceBranch
+fi
 
 dockerfile_install_gzdev_repos()
 {
 cat >> Dockerfile << DELIM_OSRF_REPO_GIT
 RUN rm -fr ${GZDEV_DIR}
-RUN git clone --depth 1 https://github.com/ignition-tooling/gzdev -b ${GZDEV_BRANCH} ${GZDEV_DIR}
+RUN git clone https://github.com/ignition-tooling/gzdev -b ${GZDEV_BRANCH} ${GZDEV_DIR}
+RUN bash -c "if [[ -n \"$GZDEV_TRY_BRANCH\" ]]; then git -C ${GZDEV_DIR} fetch origin \"$GZDEV_TRY_BRANCH\"; fi || true"
```

Comment:
This is starting to be a bit complicate to read. Do we need to use bash here or can we implement that using plan sh, something like:
```bash
RUN if [ -n foo  ]; then \
      false; \
    fi || true
```

Reply:
I think at one point I had the `"$ghprbSourceBranch" =~ ci_matching_branch\/` logic in every `RUN` statement, which requires bash, but then I moved it outside the `RUN` and forgot to revert to `sh` syntax. Thanks for noticing that; I've tried to simplify it in https://github.com/ignition-tooling/release-tools/pull/572/commits/2c6faf12af27a5caf69e208d9eceb3dfe1382e5b; let me know if it still needs more work

Reply:
wow! much better thanks!